seed/seedwriter/seed16.go

```diff
@@ -171,7 +181,14 @@ func (pol *policy16) isSystemSnapCandidate(sn *SeedSnap) bool {
 			return true
 		}
 	}
-	return pol.model.Classic() && sn.SnapName() == "core"
+	if pol.model.Classic() {
+		if pol.snapdUsedOnClassic {
+			return sn.SnapName() == "snapd"
+		} else {
+			return sn.SnapName() == "core"
```

Comment:
What is the difference in these two scenarios? Do we have classic images without snapd?